src/librustc/middle/ty.rs

```diff
@@ -1898,7 +1898,7 @@ impl<'tcx> PolyTraitRef<'tcx> {
 /// erase, or otherwise "discharge" these bound regions, we change the
 /// type from `Binder<T>` to just `T` (see
 /// e.g. `liberate_late_bound_regions`).
-#[derive(Clone, PartialEq, Eq, Hash, Debug)]
+#[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
 pub struct Binder<T>(pub T);
```

Comment:
Time to remove a thousand clone calls :P.
@huonw @Manishearth How about a lint against calling `.clone()` on a `Copy` type?